### PR TITLE
De-BCPC: Remove virt_type; Cobalt VM bits; more radosgw bits

### PR DESCRIPTION
**Pull request by cbaenziger**
This pulls out some BCPC-isms and rubocop-ifies one of the left-behind files.

Admittedly there are a ton of fixes in this branch due to me trying to test it and hitting everything under the sun (or at least our code repository):
* Issue #787 
* The datanode IPC port issue (below)
* The automated_install.sh failure to snapshot

## automated_install.sh fails to snapshot
Automated install recently got more snapshots in PR #723 but somehow the logic got flipped from what was intended. That's fixed in [this commit](https://github.com/bloomberg/chef-bach/pull/786/commits/8a130d4246e1a32e9bcdeed30ae13571925b20ad)

## Datanode IPC port issue
Another issue folks have been seeing is the install failing on bcpc-vm1 with:
Time to call off the cluster chase but I died with:
```  * bash[format namenode] action run

    ================================================================================
    Error executing action `run` on resource 'bash[format namenode]'
    ================================================================================

    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    Expected process to exit with [0], but received '1'
    ---- Begin output of "bash"  "/tmp/chef-script20170222-7332-q7qaek" ----
    STDOUT: Formatting using clusterid: CID-2c25797f-055b-4270-8c49-9b6ab1606d57
    STDERR: Java HotSpot(TM) 64-Bit Server VM warning: ignoring option PermSize=128m; support was removed in 8.0
    Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
```

I have tried to fix this with [this commit](https://github.com/bloomberg/chef-bach/pull/786/commits/9812b2cba8214edcbb04a4fd88d9101a756fe13d) thanks to @leochen4891. This is further a reasonable clean-up to prevent datanode specific settings on non-datanodes.

### CODE REVIEW

File: cookbooks/bcpc/attributes/default.rb [hunk 1]
```diff
@@ -8,17 +8,12 @@
...
 default['bcpc']['organization'] = 'Bloomberg'
 
-# Should be kvm (or qemu if testing in VMs)
-default['bcpc']['virt_type'] = 'kvm'
-
 # Region name for this cluster
 default['bcpc']['region_name'] = node.chef_environment
 
 # Domain name that will be used for DNS
 default['bcpc']['domain_name'] = 'bcpc.example.com'
```
Comment (amithkanand) [inline cookbooks/bcpc/attributes/default.rb]: Should this be removed from environment `Test-Laptop`?

File: cookbooks/bcpc/attributes/default.rb [hunk 2]
```diff
@@ -261,11 +254,6 @@
 default['bcpc']['member_role'] = 'Member'
 default['bcpc']['admin_email'] = 'admin@localhost.com'
```
Comment (amithkanand) [inline cookbooks/bcpc/attributes/default.rb]: Do we need these two attributes?

Comment (cbaenziger) [inline cookbooks/bcpc/attributes/default.rb]: `admin_email` is used by our SSL certificate recipe but I have tried to normalize some of our wonky e-mail handling in a new update.

File: cookbooks/bcpc/attributes/default.rb (hunk 1 above)
Comment (cbaenziger) [inline cookbooks/bcpc/attributes/default.rb]: Good call; I've cleaned that up and a few other duplicated and unused attributes.

File: cookbooks/bach_repository/recipes/python.rb [hunk 3]
```diff
@@ -32,12 +32,12 @@
...
 end
 
-pip_environment = {
-                   http_proxy: node['bach']['http_proxy'],
-                   https_proxy: node['bach']['https_proxy'],
-                   no_proxy: ENV['no_proxy']
-                  }
-
+pip_environment = {}
+pip_environmenet.update({http_proxy: node['bach']['http_proxy']}) if \
```
Comment (http-418) [inline cookbooks/bach_repository/recipes/python.rb]: This looks like a typo.

Comment (cbaenziger) [inline cookbooks/bach_repository/recipes/python.rb]: Ah it is; thanks for catching that!

**Comment by cbaenziger** [comment]
These commits built happily for me and setup a running HDFS cluster.

**Pull request merged**